lib/hexpm/web/controllers/dashboard_controller.ex

```diff
@@ -28,6 +28,50 @@ defmodule Hexpm.Web.DashboardController do
     end
   end
 
+  def keys(conn, _params) do
+    user = conn.assigns.current_user
+    keys = Keys.all(user)
+
+    render_keys(conn, keys)
+  end
+
+  def revoke_key(conn, %{"name" => name} = params) do
+    user = conn.assigns.current_user
+    key = Hexpm.Repo.one(Key.get(user, params["name"]))
```

Comment:
We want to avoid using Repo directly in controllers and instead use functions exposed by contexts; please use `Keys.get(user, params["name"])` instead

Reply:
@wojtekmach I'm curious to better understand the underlying reasoning behind this. Is this mainly because `Repo` leans more towards the Database/Persistence layer?

Reply:
@wojtekmach Can you please see if you could answer the above question and the question [here](https://github.com/hexpm/hexpm/pull/670#discussion_r186352220) for me ? Thank you!

Reply:
Yes, we want to hide persistence layer behind context interface.